From cf1f86663dfdbfee2036bfdca9af2d8a01f97810 Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Wed, 11 Nov 2015 18:16:48 +0530
Subject: [PATCH 01/19] FOR_UPSTREAM [VPG]: drm/i915: Avoid enabling dithering
 for 6bpc

External panels does not require dithering but was still enabled
to keep it consistent with local panels. But enabling dithering
for 6bpc on external DP will result in modifying the expected
output CRC during compliance test 4.4.1.1 to 4.4.1.3 resulting
in failing those tests. This patch fixes this by avoiding
enabling of dithering for 6bpc for external display port
panels alone.

Change-Id: Id9123bb70dde3165355a9338f5744a36bef9c1f0
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-30099
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Reviewed-on: https://android.intel.com:443/498176
---
 drivers/gpu/drm/i915/intel_display.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 58c7a05..e08bdb6 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -7208,6 +7208,17 @@ static void i9xx_set_pipeconf(struct intel_crtc *intel_crtc)
 
 	/* only g4x and later have fancy bpc/dither controls */
 	if (IS_G4X(dev) || IS_VALLEYVIEW(dev)) {
+
+		/*
+		 * compliance will fail if dithering is enabled for 6bpc
+		 * hence avoid this. This will not affect normal functioning
+		 * since external DP does not need dithering for 6bpc
+		 */
+		if (intel_pipe_has_type(&intel_crtc->base,
+				INTEL_OUTPUT_DISPLAYPORT) &&
+				intel_crtc->config.pipe_bpp < 24)
+			intel_crtc->config.dither = false;
+
 		/* Bspec claims that we can't use dithering for 30bpp pipes. */
 		if (intel_crtc->config.dither && intel_crtc->config.pipe_bpp != 30)
 			pipeconf |= PIPECONF_DITHER_EN |
-- 
1.9.1

